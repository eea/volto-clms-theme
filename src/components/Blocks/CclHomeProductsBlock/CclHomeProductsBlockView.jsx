import React from 'react';

const CclHomeProductsBlockView = (props) => {
  const { data } = props;
  let products = data?.products?.blocks_layout?.items.map(
    (uid) => data.products.blocks[uid],
  );
  return (
    <div className="home-products-container">
      {products?.map((product, index) => (
        <div
          key={index}
          className={
            product.productIcon === 'iconless'
              ? 'home-product home-product-main'
              : 'home-product'
          }
          id={product.productIcon === 'iconless' ? '' : product.productIcon}
        >
          {product.productIcon === 'iconless' ? (
            <h3>{product.title}</h3>
          ) : (
            <>
              <div className="home-product-title">{product.title}</div>
              <div className="home-product-description">
                {product.description}
              </div>
            </>
          )}
        </div>
      ))}
    </div>
  );
};

export default CclHomeProductsBlockView;
