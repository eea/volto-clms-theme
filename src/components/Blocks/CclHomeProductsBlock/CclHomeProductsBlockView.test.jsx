import React from 'react';
import renderer from 'react-test-renderer';
import CclHomeProductsBlockView from './CclHomeProductsBlockView';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const store = mockStore({
  intl: {
    locale: 'en',
    messages: {},
  },
});
describe('renders a CclHomeProductsBlockView component', () => {
  it('iconless style', () => {
    const data = {
      products: {
        blocks_layout: {
          items: ['uid1', 'uid2'],
        },
        blocks: {
          uid1: {
            title: 'example title1',
            productIcon: 'iconless',
            linkSelector: '/example',
          },
          uid2: {
            title: 'example title2',
            productIcon: 'iconless',
            linkSelector: '/example',
          },
        },
      },
    };

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <CclHomeProductsBlockView data={data} />
        </MemoryRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
  it('home-product style', () => {
    const data = {
      products: {
        blocks_layout: {
          items: ['uid1', 'uid2'],
        },
        blocks: {
          uid1: {
            title: 'example title1',
            productIcon: 'home-product',
            linkSelector: '/example',
          },
          uid2: {
            title: 'example title2',
            productIcon: 'home-product',
            linkSelector: '/example',
          },
        },
      },
    };

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <CclHomeProductsBlockView data={data} />
        </MemoryRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
