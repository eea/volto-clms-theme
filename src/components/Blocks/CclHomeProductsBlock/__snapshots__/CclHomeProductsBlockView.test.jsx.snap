// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a CclHomeProductsBlockView component home-product style 1`] = `
<div
  className="home-products-container"
>
  <a
    className="home-product"
    href="/example"
    id="home-product"
    onClick={[Function]}
    target={null}
    title={null}
  >
    <div
      className="home-product-title"
    >
      example title1
    </div>
    <div
      className="home-product-description"
    />
  </a>
  <a
    className="home-product"
    href="/example"
    id="home-product"
    onClick={[Function]}
    target={null}
    title={null}
  >
    <div
      className="home-product-title"
    >
      example title2
    </div>
    <div
      className="home-product-description"
    />
  </a>
</div>
`;

exports[`renders a CclHomeProductsBlockView component iconless style 1`] = `
<div
  className="home-products-container"
>
  <a
    className="home-product home-product-main"
    href="/example"
    id=""
    onClick={[Function]}
    target={null}
    title={null}
  >
    <h3>
      example title1
    </h3>
  </a>
  <a
    className="home-product home-product-main"
    href="/example"
    id=""
    onClick={[Function]}
    target={null}
    title={null}
  >
    <h3>
      example title2
    </h3>
  </a>
</div>
`;
